e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java

```diff
@@ -70,8 +70,21 @@ private void assertThatAccessStructuredWasFetched(String conversationId, Documen
         assertThat(accessStructured.get("taskId")).isNotNull();
     }
 
-    private void assertThatAccessDocumentWasFetched(String documentId, List<Document> accessDocuments) {
-        Document document = accessDocuments.get(0);
+    private Document theDocumentTaskUpdatesTheRecord(String conversationId) {
+        var gpcAccessDocument = (Document) Mongo.findEhrExtractStatus(conversationId).get(GPC_ACCESS_DOCUMENT);
+        if(gpcAccessDocument != null) {
+            var documentList = (List) gpcAccessDocument.get("documents");
+            if(documentList != null && !documentList.isEmpty()) {
```

Comment:
We could use CollectionUtils to check if our list is empty.

Reply:
not a current dependency of this project, will leave as-is

Reply:
Ok

Reply:
re-wrote to avoid null check and class cast